### DISCUSSION TITLE
Image upload type in Traits

### DISCUSSION THREAD

**Discussion by zeeshan-algo** [GrapesJS/grapesjs#4932] [Q&A] [answered]
I want to add type image upload  like in this picture . What i want is to upload an image from traits and it show on the canvas(component).How can i achieve this.

![practice area](https://user-images.githubusercontent.com/115087247/219308984-b6bc919d-1c06-4b05-888c-dea32a65740e.png)

**Comment by zeeshan-algo** [comment]
Here is my code 
`  editor.DomComponents.addType('image', {
      model: {
        defaults: {
          traits: [
            {
              type: 'file',
              name: 'src',
              label: 'Image',
              accept: 'image/*',
              changeProp: 1,
              value: '',
              options: {
                onSelect: function (value) {
                  var file = value.file;
                  var reader = new FileReader();
                  reader.onload = function () {
                    // @ts-ignore
                    imageTrait.set('value', reader.result);
                  };
                  reader.readAsDataURL(file);
                },
              },
            },
          ],
        },
      },
    });`

**Comment by artf** [comment, marked as answer]
Hi @zeeshan-algo there is no built-in `file` type in Traits, but you can define your own as described [here](https://grapesjs.com/docs/modules/Traits.html#define-new-trait-type) with your UI and logic.
